### PR TITLE
4.08 fixes and conduit-mirage support and use a logging function 

### PR DESCRIPTION
**Pull request by avsm**

### CODE REVIEW

File: cohttp-async/bin/cohttp_curl_async.ml [hunk 1]
```diff
@@ -31,7 +31,7 @@ let make_net_req uri meth' body () =
   show_headers (Cohttp.Response.headers res);
   body
   |> Body.to_pipe
-  |> Pipe.iter ~f:(fun b -> Caml.Pervasives.print_string b; return ())
+  |> Pipe.iter ~f:(fun b -> Stdlib.print_string b; return ())
```
Comment (mseri) [inline cohttp-async/bin/cohttp_curl_async.ml]: Doesn’t this also require to depend on `stdlib-shims` for `cohttp-async`?

**Comment by mseri** [comment]
```
The following dependencies couldn't be met:
 - cohttp-mirage -> conduit-mirage
      unknown package
```

**Comment by mseri** [comment]
Oh I just noticed the new PR to release conduit, sorry.

File: cohttp-async/bin/cohttp_curl_async.ml (hunk 1 above)
Comment (avsm) [reply cohttp-async/bin/cohttp_curl_async.ml]: It does! Good catch, fixing now.

**Comment by avsm** [comment]
Yeah, this PR will pass once ocaml/opam-repository#14689 is merged

**Pull request merged**